subprojects/core/src/main/java/org/gradle/api/internal/file/UnionFileCollection.java

```diff
@@ -42,10 +42,8 @@ public String getDisplayName() {
         return source;
     }
 
-    @Override
-    public FileCollection add(FileCollection collection) {
+    public void uniteWith(FileCollection collection) {
```

Comment:
❌ Aren't we breaking clients using the add method on `UnionFileCollection` since we now throw an `UnsupportedOperationException` instead of adding to the collection?

I think we should print a deprecation warning while still calling `uniteWith`.

Reply:
Good point.